PipelineCaseIterator/PipelineCaseIterator.py

```diff
@@ -4,300 +4,493 @@
 import re
 import subprocess
 import traceback
-
-from Widgets.SelectModulePopUp import SelectModulePopUp
+import typing
+import csv
+
+from slicer.parameterNodeWrapper import (
+    isParameterPack,
+    splitAnnotations,
+    unannotatedType,
+)
+from Widgets.SelectPipelinePopUp import SelectPipelinePopUp
+# SelectModulePopUp
+from PipelineCreator import slicerPipeline
+from _PipelineCreator.PipelineRegistrar import PipelineInfo
 import vtk, qt, slicer
 from slicer.ScriptedLoadableModule import *
 from slicer.util import VTKObservationMixin
-from PipelineCreator import PipelineCreatorLogic
-from PipelineCreatorLib.Asynchrony import Asynchrony
-from PipelineModulesLib.Util import ScopedNode, ScopedDefaultStorageNode, human_sorted
+from PipelineCreator import PipelineCreatorLogic, PipelineProgressCallback
+from PipelineCaseIteratorLibrary import *
```

Comment:
This fails on cmake install since `PipelineCaseIteratorLibrary/__init__.py` isn't included in `PipelineCaseIterator/CMakeLists.txt`, so `import *` grabs the child modules instead.

Leads to strange NameErrors later on in the code, `Asynchrony` and `IteratorParameterFile` are missing.

I suggest adding an `__all__` to `PipelineCaseIteratorLibrary/__init__.py` and replacing these `import *` usages with explicit lists as in lines 10-14 to more easily catch this kind of error in the future.

Reply:
I pushed 6d44c90 for those first two points, but I left `import *` in since that's more invasive and I'm not entirely sure where all the usages of those classes are.

For reference, the steps to reproduce the issue:

```
gh repo clone KitwareMedical/SlicerPipelines
cd SlicerPipelines
cmake -B build
cmake --build build
build/SlicerWithPipelines
```

Then follow the guide you listed in https://github.com/KitwareMedical/SlicerPipelines/pull/23#issuecomment-1679488895